lxd/network/network_utils.go

```diff
@@ -55,6 +55,15 @@ func validRealNetworkName(value string) error {
 	return nil
 }
 
+// validVirtualNetworkName validates a virtual network name (one that doesn't have an actual network interface).
+func validVirtualNetworkName(value string) error {
+	if strings.Contains(value, "/") {
+		return fmt.Errorf(`Network name cannot contain "\"`)
```

Comment:
Hmm, error is incorrect, should say `/`

Reply:
Ah yes I converted it from being escaped and took the wrong slash.

Reply:
@stgraber is the reason we don't allow `/` in these names is to prevent issues when using the name in the API URL segments? If so should we also ban things like `?` and `&` or should we just update the client to always path escape these values, using https://golang.org/pkg/net/url/#PathEscape?

Reply:
Yeah, `/` is a bit worse than the others even with encoding/decoding.

Though I think we should implement a generic validator for those names that only ever make it in API URLs (so don't have filesystem/OS constraints) and so apply consistent rules across the board.

I'd certainly be happy to ban `/`, `?` and `&` for all of those as that makes it much harder for someone to deal with in URLs and those also aren't normal separators a user is likely to use anyway (well, `/` is but we're already blocking that).